### PR TITLE
updated with ++/-- functionality

### PR DESCRIPTION
**Pull request by travisdesell**
Might not want to merge anything that's not leaderboard.rb

### CODE REVIEW

**Comment by benjamincudi** [comment]
But that's not how merges work, :-1 for manually merging

File: lib/slack_config.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def set_config(config_variable, default = nil, config_file_index = nil)
       config_variable = config_variable.to_s
       if (config_variable == "ssl")
         @config.ssl.use = file_config(config_file_index, false)
-        @config.ssl.verify = file_config(config_file_index, false)
+        #@config.ssl.verify = file_config(config_file_index, false)
```
Comment (benjamincudi) [inline lib/slack_config.rb]: Probably want to reverse this

File: slackcat.rb [hunk 2]
```diff
@@ -3,87 +3,67 @@
...
-  set_config :server
-  set_config :port, 6667
-  load_plugins
+    set_config :ssl, true
+    set_config :nick, "travisbot"
+    set_config :user, "travisbot"
+    set_config :password
+    set_config :server, "roguesquadron.irc.slack.com"
+    set_config :port, 6697
+    #set_config :plugins, (["leaderboard"].to_s)
```
Comment (benjamincudi) [inline slackcat.rb]: Probably want to reverse this config block change

File: slackcat.rb [hunk 3]
```diff
@@ -3,87 +3,67 @@
...
+    #    name = m.message.split("+")[0]
+    #    info("incrementing '#{name}'")
+    #    m.reply "incremeting '#{name}'! - suck it jpaul"
+    #end
+
+    #on :message, /[a-zA-Z\.]--$/ do |m|
+    #    name = m.message.split("-")[0]
+    #    info("decrementing'#{name}'")
+    #    m.reply "decrementing '#{name}'! - suck it jpaul"
+    #end
```
Comment (benjamincudi) [inline slackcat.rb]: Guessing this is cruft from your first pass at implementing, would be worth deleting

File: slackcat.rb [hunk 4]
```diff
@@ -3,87 +3,67 @@
...
+    on :message, /;part (.+)/ do |m|
+        channel = m.message.split(" ")[1]
+        info("trying to part channel: #{channel}")
+        bot.part(channel)
     end
-  end
 
-  command /source/ do |message|
-    message.reply "https://github.com/jayson/SlackCat"
-  end
```
Comment (benjamincudi) [inline slackcat.rb]: Why would you remove this?

File: slackcat.rb [hunk 5]
```diff
@@ -3,87 +3,67 @@
...
-      until out.eof?
-        # raise "Timeout" if output.empty? && Time.now.to_i - start > 300
-        chr = out.read(1)
-        output << chr
-      end
-      error_message = nil
-      error_message = err.read unless err.eof?
-      message.reply("#{output} #{error_message}")
-    end
-  end
```
Comment (benjamincudi) [inline slackcat.rb]: Another arbitrarily removed function?

File: slackcat.rb [hunk 6]
```diff
@@ -3,87 +3,67 @@
...
+        info("trying to join channel: #{channel}")
+        bot.join(channel)
     end
-  end
 
-  command /reload/ do |message|
-    load_config
-    @slack_config = JSON.parse(File.read(".config"))
-    @slack_config["channels"].each do |channel| 
-      bot.join(channel)
```
Comment (benjamincudi) [inline slackcat.rb]: Looks like `reload` was lost as well

**Comment by jayson** [comment]
Refactored entirely and merged manually here: https://github.com/jayson/SlackCat/compare/63b9953bc0b641271e1602cffd2b7b4db723dc2c...c1a59e34aa772a7f53b8318a75e06ede66bae298

**Pull request closed without merging**